### PR TITLE
Public Cloud: Terraform: Keep current providers version and fix the file syntax

### PR DESCRIPTION
**Pull request by pdostal**
This is to protect our Terraform Public Cloud workflow so we providers of specific version.

- Related ticket: [poo#91944](https://progress.opensuse.org/issues/91944)
- Verification run: [AWS](http://pdostal-server.suse.cz/tests/11784) [Azure](http://pdostal-server.suse.cz/tests/11785) [Google](http://pdostal-server.suse.cz/tests/11786)

### CODE REVIEW

File: data/publiccloud/terraform/azure.tf [hunk 1]
```diff
@@ -1,3 +1,12 @@
+terraform {
+  required_providers {
+    azurerm = {
+      version = "= 2.56.0"
```
Comment (asmorodskyi) [inline data/publiccloud/terraform/azure.tf:4]:
```
Initializing provider plugins...
- Finding latest version of hashicorp/random...
- Finding latest version of hashicorp/azurerm...
- Installing hashicorp/random v3.1.0...
- Installed hashicorp/random v3.1.0 (self-signed, key ID 34365D9472D7468F)
- Installing hashicorp/azurerm v2.57.0...
- Installed hashicorp/azurerm v2.57.0 (self-signed, key ID 34365D9472D7468F)
- ```
- I would also lock `random` module

Comment (asmorodskyi) [reply data/publiccloud/terraform/azure.tf:4]: I still see this in your VR. 
```
Warning: Interpolation-only expressions are deprecated

  on plan.tf line 69, in resource "azurerm_resource_group" "openqa-group":
  69:             openqa_created_date = "${timestamp()}"
```

I think we need to solve this anyway . Or do you think that it is better to do this in separate PR ?

Comment (pdostal) [reply data/publiccloud/terraform/azure.tf:4]: Good ideas, I will do solve both.

File: data/publiccloud/terraform/ec2.tf [hunk 2]
```diff
@@ -1,3 +1,12 @@
+terraform {
+  required_providers {
+    aws = {
+      version = "= 3.37.0"
```
Comment (asmorodskyi) [inline data/publiccloud/terraform/ec2.tf:4]: both ideas applicable for AWS too

**Comment by pdostal** [comment]
@asmorodskyi please re-check.

**Comment by asmorodskyi** [comment]
code looks good , what about VRs ?

**Comment by pdostal** [comment]
> code looks good , what about VRs ?

Description updated, just waiting for Azure but it looks good.

**Comment by asmorodskyi** [comment]
```
Warning: Interpolation-only expressions are deprecated

  on plan.tf line 98, in resource "aws_instance" "openqa":
  98:     security_groups = ["${aws_security_group.basic_sg.name}"]
  ```
in last ec2 run

**Comment by asmorodskyi** [comment]
gce has one more plugin : 
```
Initializing provider plugins...
- Finding hashicorp/random versions matching "3.1.0"...
- Finding hashicorp/google versions matching "3.65.0"...
- Finding latest version of hashicorp/external...
- Installing hashicorp/google v3.65.0...
- Installed hashicorp/google v3.65.0 (self-signed, key ID 34365D9472D7468F)
- Installing hashicorp/external v2.1.0...
- Installed hashicorp/external v2.1.0 (self-signed, key ID 34365D9472D7468F)
- Installing hashicorp/random v3.1.0...
- Installed hashicorp/random v3.1.0 (self-signed, key ID 34365D9472D7468F)
```

**Comment by pdostal** [comment]
Thank you @asmorodskyi, please check once more.

**Pull request merged**